### ISSUE TITLE
Catalog updated before data processing

### ISSUE THREAD

**Issue by dongbohu** [greenelab/cimr#42]
According to: https://github.com/greenelab/cimr/blob/12865829db3fdcd1c85e1ad187f073c38a69e72f/cimr/processor/yamler.py#L560
the catalog file seems to be updated **BEFORE** data processing starts. I think it makes more sense to update it **AFTER** data is successfully processed.

**Comment by dongbohu** [comment]
I ran a test and confirmed that the catalog was updated even when data processing was terminated due to some errors.

**Comment by ypar** [comment]
Yes, but nothing is committed until the processing is done so it does not effect any outcome either way. Regardless, I can take a look.
